### PR TITLE
Adds bit manipulations over byte openarrays

### PR DESCRIPTION
**Pull request by zah**

### CODE REVIEW

File: ranges/bitranges.nim [hunk 1]
```diff
@@ -123,6 +130,12 @@ iterator enumerateBits(x: BitRange): (int, bool) =
     inc p
     inc i
 
+proc getBit*(bytes: openarray[byte], pos: Natural): bool =
+  getAbsoluteBit(bytes, pos)
+
+proc setBitBE*(bytes: var openarray[byte], pos: Natural, value: bool) =
```
Comment (arnetheduck) [inline ranges/bitranges.nim]: ugh, why is `BE` in the name here and not everywhere else?

Comment (arnetheduck) [reply ranges/bitranges.nim]: ah never mind, just saw that the code is littered with these endianess concerns.. looks a bit out of place - a byte array does not really have an endianness, unless it's bit-endianess which for sure would be unusual

**Comment by arnetheduck** [comment]
@zah wait, what? Big endian as default?? that's pretty unusual - the normal thing to do is that bits are set by a `1 << n` expression - just like that article points out.

**Comment by zah** [comment]
Oh, right, crap! The binary Trie code is then currently full of bit indexing with `bits[n]` that uses the non-default treatment and will have to be reworked.

**Pull request merged**